caikit_nlp/toolkit/torch_run.py

```diff
@@ -94,4 +94,5 @@ def get_torch_elastic_launch_config(
         rdzv_endpoint=f"{master_addr}:{master_port}",
         rdzv_configs=rdzv_configs,
         tee=Std.ALL,
+        max_restarts=max_restarts,
     )
```

Comment:
Curious, did we see any failure which required to add restart counts?

Reply:
Yup, unfortunately. If the batch size is too large, it seems to cause some issues for normal style datasets.

Example:
```
ALLOW_DOWNLOADS=true MASTER_ADDR=localhost MASTER_PORT=25580 WORLD_SIZE=2 RANK=0 LOGLEVEL=DEBUG python run_fine_tuning.py --dataset "billsum"  --model_name t5-small --num_epochs 1 --output_dir tmp/with_loss_log --accumulate_steps 16 --max_source_length 512 --metric rouge --torch_dtype float16 --batch_size=256
```

it will crash the subprocess with a tensor dimension error, which ultimately leads to restarts. 

```
 RuntimeError: The size of tensor a (8225280) must match the size of tensor b (16450560) at non-singleton dimension 0
```
This is a little annoying because it's a slow error, and it also doesn't log the error to the console until the restarts have tried and failed